erpnext/payroll/doctype/employee_incentive/employee_incentive.js

```diff
@@ -11,12 +11,57 @@ frappe.ui.form.on('Employee Incentive', {
 			};
 		});
 
+		if(!frm.doc.currency) return;
 		frm.set_query("salary_component", function() {
 			return {
-				filters: {
-					"type": "Earning"
-				}
+				query : "erpnext.payroll.doctype.salary_structure.salary_structure.get_earning_deduction_components",
+				filters: {type: "earning", currency: frm.doc.currency, company: frm.doc.company}
 			};
 		});
-	}
+
+	},
+
+	employee: function(frm) {
+		if (frm.doc.employee) {
+			frappe.run_serially([
+				() => 	frm.trigger('get_employee_currency'),
+				() => 	frm.trigger('set_company')
+			]);
+		} else {
+			frm.set_value("company", null);
+		}
+	},
+
+	set_company: function(frm) {
+		frappe.call({
+			method: "frappe.client.get_value",
+			args:{
+				doctype: "Employee",
+				fieldname: "company",
+				filters:{
+					name: frm.doc.employee
+				}
+			},
+			callback: function(data) {
+				if(data.message){
+					frm.set_value("company", data.message.company);
+				}
+			}
+		});
+	},
+
+	get_employee_currency: function(frm) {
+		frappe.call({
+			method: "erpnext.payroll.doctype.salary_structure_assignment.salary_structure_assignment.get_employee_currency",
```

Comment:
set currency dynamically based on the currency field.
![image](https://user-images.githubusercontent.com/32095923/99652345-b91e9580-2a7d-11eb-88f0-40847440dac9.png)

Reply:
It doesn't show up until it is saved or there is another currency field and you start filling it.